python/src/scipp/compat/mantid.py

```diff
@@ -254,13 +254,25 @@ def set_bin_masks(bin_masks, dim, index, masked_bins):
 
 
 def _convert_MatrixWorkspace_info(ws):
+    common_bins = ws.isCommonBins()
+    dim, unit = validate_and_get_unit(ws.getAxis(0).getUnit().unitID())
     source_pos, sample_pos = make_component_info(ws)
     det_info = make_detector_info(ws)
     pos = init_pos(ws)
     spec_dim, spec_coord = init_spec_axis(ws)
 
+    if common_bins:
+        coord = sc.Variable([dim], values=ws.readX(0), unit=unit)
+    else:
+        coord = sc.Variable([spec_dim, dim],
+                            shape=(ws.getNumberHistograms(), len(ws.readX(0))),
```

Comment:
I think the assumption here is that each spectrum has the same number of bins. I believe this is still valid, but I remember Pete discussing this a while back. 

You could also call `ws.blocksize()` to do the same as `readX(0)` here.

Reply:
I think you are right that this is not true in general anymore. Fixed.

Reply:
... actually no, because we need the length o the bin edges, not the data length, which is what `blocksize()` gives. I think we can continue realying on failing due to length mismatches elsewhere.